crates/core/src/transcript/proof.rs

```diff
@@ -206,50 +285,77 @@ impl<'a> TranscriptProofBuilder<'a> {
                     ));
                 };
 
-                let dir_idx = (direction, idx);
-
-                if !encoding_tree.contains(&dir_idx) {
-                    return Err(TranscriptProofBuilderError::new(
-                        BuilderErrorKind::MissingCommitment,
-                        format!(
-                            "encoding commitment is missing for ranges in {} transcript",
-                            direction
-                        ),
-                    ));
+                // Insert the rangeset if it's in the encoding tree (which means it's a
+                // committed rangeset).
+                if encoding_tree.contains(&dir_idx) {
+                    self.encoding_proof_idxs.insert(dir_idx);
+                } else {
+                    let mut missing_commitment = true;
+                    // Check if there is any committed rangeset in the encoding tree that is a
+                    // subset of the rangeset — if yes, insert them.
+                    for committed_dir_idx in encoding_tree
+                        .transcript_indices()
+                        .into_iter()
+                        .filter(|(dir, _)| *dir == dir_idx.0)
+                    {
+                        if committed_dir_idx.1.is_subset(&dir_idx.1) {
+                            self.encoding_proof_idxs.insert(committed_dir_idx.clone());
+                            missing_commitment = false;
+                        }
+                    }
+                    // If no committed rangeset is a subset, that means the rangeset is missing
+                    // fully or partially in the encoding tree (which means it has not been
+                    // committed).
+                    if missing_commitment {
```

Comment:
@yuroitaki , a quick observation here (I haven't fully grokked the rest of the logic, so maybe this comment is irrelevant).

Assume there exists a proof in the `encoding_tree` for ranges 0..20. Assume the prover calls reveal(0..25). 
The current logic will insert 0..20 into `encoding_proof_idxs`and will set `missing_commitment` to false.

I think we should make it an error - if the user tries to reveal something which was not committed, we want to error out.

Reply:
@themighty1 interesting, I was assuming`(0..25)` to be treated as the umbrella/superset range, but if it only reveals a *single* subset range, then it doesn't achieve the purpose of this PR: to support revealing *multiple* subset ranges. I will make the change.

Reply:
Upon clarification, we agreed on making the reveal API to only support revealing the exact rangeset that's passed in. For example, if two rangesets are committed: `[0..2, 8..9], [2..4, 9..11]`, then

(a) revealing `[0..4, 8..11]` is valid
(b) revealing `[0..11]` is not valid